### PR TITLE
RDX: Implement fetching from backend

### PR DESCRIPTION
**Pull request by mook-as**
Implements #4402.  The commits should be reasonably isolated, so reviewing commit-by-commit might be easier.

### CODE REVIEW

File: scripts/lib/build-utils.ts [hunk 1]
```diff
@@ -295,6 +312,112 @@ export default {
...
+
+      // Build the golang executable
+      await this.spawn('go', 'build', '-ldflags', '-s -w', '-o', executablePath, '.', {
+        cwd: path.join(this.rootDir, 'src', 'go', 'extension-proxy'),
+        env: { ...process.env, GOOS: 'linux' },
+      });
+
+      // Build the layer tarball
+      // tar streams don't implement piping to multiple writers, and stream.Duplex
+      // can't deal with it either; so we need to fully writ out the file, then
```
Comment (ericpromislow) [inline scripts/lib/build-utils.ts:333]: typo: "write"

File: scripts/lib/build-utils.ts [hunk 2]
```diff
@@ -295,6 +312,112 @@ export default {
...
+      // calculate the hash as a separate step.
+      const layer = tar.pack();
```
Comment (ericpromislow) [inline scripts/lib/build-utils.ts:335]: I don't see this tar API at https://www.npmjs.com/package/tar -- could you add a link pointing to the API you're using in this module?

Comment (mook-as) [reply scripts/lib/build-utils.ts:335]: I'm using https://www.npmjs.com/package/tar-stream — see the top of the file where it's imported?

File: pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts [hunk 3]
```diff
@@ -431,6 +431,45 @@ export class NerdctlClient implements ContainerEngineClient {
...
+      const args = [
+        options.namespace ? ['--namespace', options.namespace] : [],
+        ['compose'],
+        options.name ? ['--project-name', options.name] : [],
+        ['--project-directory', workComposeDir],
+      ].flat();
+
+      if (options.env) {
+        const envFile = (await this.vm.execCommand({ capture: true },
+          '/bin/mktemp', '--tmpdir', 'rd-nerdctl-compose-exec-XXXXXX')).trim();
```
Comment (ericpromislow) [inline pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts:450]: I guess there's no need to clean this up on the VM

File: pkg/rancher-desktop/main/extensions/manager.ts [hunk 4]
```diff
@@ -168,7 +168,21 @@ class ExtensionManagerImpl implements ExtensionManager {
 
     this.setMainHandler('extensions/vm/http-fetch', async(event, config) => {
       const extensionId = this.getExtensionIdFromEvent(event);
-      const url = new URL(config.url);
+      const extension = this.getExtension(extensionId) as ExtensionImpl;
+      let url: URL;
+
+      if (/^[^:/]*:/.test(config.url)) {
+        // the URL is absolute, use as-is.
```
Comment (ericpromislow) [inline pkg/rancher-desktop/main/extensions/manager.ts:175]: This pattern looks to me like it's matching the protocol or domain of a URL, which I suppose makes it absolute.

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Just some at-the-micro level questions.

I read the commits in order as recommended and this makes sense otherwise.

**Comment by ericpromislow** [comment]
This also left behind `resources/rdx-proxy.tgz` - should it be added to `.gitignore`?

File: scripts/lib/build-utils.ts (hunk 2 above)
Comment (ericpromislow) [reply scripts/lib/build-utils.ts:335]: Sorry, I searched npm before checking the import statements

File: pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts (hunk 3 above)
Comment (mook-as) [reply pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts:450]: The next line does the cleanup (pushing the `rm` into `cleanups`, which get run at the end of the function in a `finally`).

I'd have preferred `defer`, but that's a golang thing and not a JavaScript thing.

File: pkg/rancher-desktop/main/extensions/manager.ts (hunk 4 above)
Comment (mook-as) [reply pkg/rancher-desktop/main/extensions/manager.ts:175]: Yep.

**Review by mook-as — COMMENTED** [review]
The `.gitignore` was updated in 17066e75fbc3b67dda603c511ec2ddf9f4fdfb44 where we started building `/resources/rdx-proxy.tgz`.

**Review by ericpromislow — DISMISSED** [review]
looks fine

**Review by ericpromislow — APPROVED** [review]
Verified that commit `RDX: build extension proxy without CGO` fixes the linux problem

**Pull request merged**